### PR TITLE
fix(treesitter): escape all '}' in require_language

### PR DESCRIPTION
**Pull request by rlwrnc**
Fix for #21485.

### CODE REVIEW

**Comment by clason** [comment]
Please explain _what_ this fixes.

**Comment by rlwrnc** [comment]
> Please explain _what_ this fixes.
Apologies. Seems to fix #21485.

**Comment by clason** [comment]
Please add the explanation to the PR description and commit message.

File: runtime/lua/vim/treesitter/language.lua [hunk 1]
```diff
@@ -16,7 +16,9 @@ function M.require_language(lang, path, silent, symbol_name)
     return true
   end
   if path == nil then
+    local extra_escape_chars = '}'
     local fname = 'parser/' .. vim.fn.fnameescape(lang) .. '.*'
```
Comment (justinmk) [inline runtime/lua/vim/treesitter/language.lua:20]: escape `lang`, don't escape `fname`

**Pull request closed without merging**